### PR TITLE
chore(tests): unbreak tests on Windows (#1775) #1837

### PR DESCRIPTION
**Pull request by joshua-goldstein**
Cherry pick of #1775 and #1837 .

### CODE REVIEW

File: value_test.go [hunk 1]
```diff
@@ -724,8 +725,12 @@ func TestReadOnlyOpenWithPartialAppendToWAL(t *testing.T) {
 	opts.ReadOnly = true
 	// Badger should fail a read-only open with values to replay
 	_, err = Open(opts)
-	require.Error(t, err)
-	require.Regexp(t, "Log truncate required", err.Error())
+	if runtime.GOOS == "windows" {
+		require.Equal(t, err, ErrWindowsNotSupported)
```
Comment (mangalaman93) [inline value_test.go:729]: could you just return after this line? and not have the else block

File: db_test.go [hunk 2]
```diff
@@ -2279,10 +2285,14 @@ func TestOpenDBReadOnly(t *testing.T) {
 
 	ops.ReadOnly = true
 	db, err = Open(ops)
-	require.NoError(t, err)
-	read()
-	require.Equal(t, 20, count)
-	require.NoError(t, db.Close())
+	if runtime.GOOS == "windows" {
+		require.Equal(t, err, ErrWindowsNotSupported)
```
Comment (mangalaman93) [inline db_test.go:2289]: same here, return after this line and we don't need the else line

File: db_test.go [hunk 3]
```diff
@@ -2234,11 +2234,13 @@ func TestOpenDBReadOnly(t *testing.T) {
 
 	ops.ReadOnly = true
 	db, err = Open(ops)
-	require.NoError(t, err)
-	require.NoError(t, db.Close())
+	if runtime.GOOS == "windows" {
+		require.Equal(t, err, ErrWindowsNotSupported)
```
Comment (mangalaman93) [inline db_test.go:2238]: same here

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

File: value_test.go (hunk 1 above)
Comment (joshua-goldstein) [reply value_test.go:729]: done

**Comment by mangalaman93** [comment]
LGTM. Just one more thing, can we add a comment explaining why things break on windows? And how do we know that the tests now pass on windows?

**Comment by mangalaman93** [comment]
One more question, do we run CI on windows?

**Pull request closed without merging**